lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -128,12 +128,17 @@ FunctionType rtl::getModuleType(Operation *module) {
 
 /// Return the port name for the specified argument or result.
 StringAttr rtl::getModuleArgumentNameAttr(Operation *module, size_t argNo) {
-  return module->getAttrOfType<ArrayAttr>("argNames")[argNo].cast<StringAttr>();
+  auto arguments = module->getAttrOfType<ArrayAttr>("argNames");
+  if (!arguments || argNo >= arguments.size())
+    return {};
+  return arguments[argNo].cast<StringAttr>();
```

Comment:
Please don't do this.  The verifier intentionally checks that argNames/resultNames are the right length.  It isn't proper to relax invariants like this, please put the complexity in the lowering pass.